Src/Support/Google.Apis.Auth/OAuth2/AccessTokenWithHeaders.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+Copyright 2019 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Represents an access token that can be used to authorize a request.
+    /// The token might be accompanied by extra information that should be sent
+    /// in the form of headers.
+    /// </summary>
+    public class AccessTokenWithHeaders
```

Comment:
Please can this class be changed to be immutable?
It only ever has its properties set at construction time, and there will not be any need for them to ever be mutated.
(Some of the other comments on this class will become irrelevant if this change is made!)

Reply:
Yes I doubted some about this. Would you be OK with an  AccessTokenWithHeadersBuilder class to allow adding headers at other than construction. I know that we don't necessarily need that right now...